services/server/src/server/services/workers/verificationWorker.ts

```diff
@@ -0,0 +1,158 @@
+import Piscina from "piscina";
+import {
+  SolidityJsonInput,
+  VyperJsonInput,
+  CompilationTarget,
+  SolidityCompilation,
+  VyperCompilation,
+  Verification,
+  SourcifyLibError,
+  SourcifyChain,
+  SourcifyChainInstance,
+  SourcifyChainMap,
+} from "@ethereum-sourcify/lib-sourcify";
+import { resolve } from "path";
+import { ChainRepository } from "../../../sourcify-chain-repository";
+import { SolcLocal } from "../compiler/local/SolcLocal";
+import { VyperLocal } from "../compiler/local/VyperLocal";
+import {
+  getVerificationErrorMessage,
+  MatchingErrorResponse,
+} from "../../apiv2/errors";
+import { v4 as uuidv4 } from "uuid";
+import { getCreatorTx } from "../utils/contract-creation-util";
+import { VerifyFromJsonInputs, VerifyFromJsonOutput } from "./workerTypes";
+
+export const filename = resolve(__filename);
+
+let chainRepository: ChainRepository;
+let solc: SolcLocal;
+let vyper: VyperLocal;
+
+const initWorker = () => {
+  if (chainRepository && solc && vyper) {
+    return;
+  }
+
+  const sourcifyChainInstanceMap = Piscina.workerData
+    .sourcifyChainInstanceMap as { [chainId: string]: SourcifyChainInstance };
+
+  const sourcifyChainMap = Object.entries(sourcifyChainInstanceMap).reduce(
+    (acc, [chainId, chain]) => {
+      acc[chainId] = new SourcifyChain(chain);
+      return acc;
+    },
+    {} as SourcifyChainMap,
+  );
+
+  chainRepository = new ChainRepository(sourcifyChainMap);
+  solc = new SolcLocal(
+    Piscina.workerData.solcRepoPath,
+    Piscina.workerData.solJsonRepoPath,
+  );
+  vyper = new VyperLocal(Piscina.workerData.vyperRepoPath);
+};
+
+export async function verifyFromJsonInput({
+  chainId,
+  address,
+  jsonInput,
+  compilerVersion,
+  contractIdentifier,
+  creationTransactionHash,
+}: VerifyFromJsonInputs): Promise<VerifyFromJsonOutput> {
+  initWorker();
+
+  // The contract path can include a colon itself,
+  // therefore we need to take the last element as the contract name
+  const splitIdentifier = contractIdentifier.split(":");
+  const contractName = splitIdentifier[splitIdentifier.length - 1];
+  const contractPath = splitIdentifier.slice(0, -1).join(":");
+  const compilationTarget: CompilationTarget = {
+    name: contractName,
+    path: contractPath,
+  };
```

Comment:
Can we actually handle this case from outside the worker? It feels to me that having clear inputs for the worker is somehow better

Reply:
Yes, I think that makes sense.